src/webui/www/private/scripts/mocha-init.js

```diff
@@ -1187,22 +1187,63 @@ const initializeWindows = () => {
 
     exportTorrentFN = async () => {
         const hashes = torrentsTable.selectedRowsIds();
-        for (const hash of hashes) {
+
+        if (hashes.length === 0)
+            return;
+
+        if (hashes.length === 1) {
+            // Single file - direct download
+            const hash = hashes[0];
             const row = torrentsTable.getRow(hash);
             if (!row)
-                continue;
+                return;
 
             const name = row.full_data.name;
             const url = new URL("api/v2/torrents/export", window.location);
-            url.search = new URLSearchParams({
-                hash: hash
+            url.search = new URLSearchParams({ hash });
+
+            await window.qBittorrent.Misc.downloadFile(
+                url,
+                `${name}.torrent`,
+                "QBT_TR(Unable to export torrent file)QBT_TR[CONTEXT=MainWindow]"
+            );
+        }
+        else {
+            // Multi-file - distributed in a zip
+            const zip = new JSZip();
+
+            const fetchTorrents = hashes.map(async (hash) => {
+                const row = torrentsTable.getRow(hash);
+                if (!row)
+                    return;
+
+                const name = row.full_data.name;
+                const url = new URL("api/v2/torrents/export", window.location);
+                url.search = new URLSearchParams({ hash });
+
+                try {
+                    const response = await fetch(url);
+                    if (!response.ok)
+                        throw new Error();
+
+                    const blob = await response.blob();
+                    zip.file(`${name}.torrent`, blob);
+                }
+                catch (error) {
+                    alert("QBT_TR(Unable to export torrent file)QBT_TR[CONTEXT=MainWindow]");
+                }
```

Comment:
This will produce many alerts when a batch goes wrong. Do you intend the user to face all of them? Why don't you handle the errors at `Promise.all`?